x-pack/plugins/remote_clusters/common/lib/cluster_serialization.ts

```diff
@@ -157,7 +157,7 @@ export function serializeCluster(deserializedClusterObject: Cluster): ClusterPay
             mode: mode ?? null,
             proxy_address: proxyAddress ?? null,
             proxy_socket_connections: proxySocketConnections ?? null,
-            server_name: serverName ?? null,
+            server_name: serverName && serverName !== '' ? serverName : null,
```

Comment:
This part: `&& serverName !== ''` will not be evaluated if `serverName` is `''` because `''` is already falsey. It looks like the fix is `serverName ? serverName : null` where the `??` was too restrictively looking at only overriding `null` or `undefined`.

Do you we face a similar challenge with the other values here?

Reply:
Or just `serverName || null`

Reply:
🤦‍♀ you're correct! I updated the code. I think it makes sense to use `<value> || null` for all values as any falsely value would be invalid.